presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitPartitionInfo.java

```diff
@@ -45,16 +50,44 @@
 
     public HiveSplitPartitionInfo(
             Properties schema,
+            URI path,
             List<HivePartitionKey> partitionKeys,
             String partitionName,
             Map<Integer, HiveTypeName> columnCoercions,
             Optional<HiveSplit.BucketConversion> bucketConversion)
     {
-        this.schema = requireNonNull(schema, "schema is null");
-        this.partitionKeys = requireNonNull(partitionKeys, "partitionKeys is null");
-        this.partitionName = requireNonNull(partitionName, "partitionName is null");
-        this.columnCoercions = requireNonNull(columnCoercions, "columnCoersions is null");
-        this.bucketConversion = requireNonNull(bucketConversion, "bucketConversion is null");
+        requireNonNull(schema, "schema is null");
+        requireNonNull(path, "path is null");
+        requireNonNull(partitionKeys, "partitionKeys is null");
+        requireNonNull(partitionName, "partitionName is null");
+        requireNonNull(columnCoercions, "columnCoersions is null");
+        requireNonNull(bucketConversion, "bucketConversion is null");
+
+        this.schema = schema;
+        this.path = ensurePathHasTrailingSlash(path);
```

Comment:
e.g.: `this.path = ensurePathHasTrailingSlash(requireNonNull(path, "path is null"));`